generator/sources/java/src/com/kaltura/client/KalturaClientBase.java

```diff
@@ -177,28 +184,48 @@ protected InputStream getResponseBody(PostMethod postMethod)
     
     public Header[] getResponseHeaders()
     {
-        return responseHeaders;
+    	if(threadsInfo.containsKey(Thread.currentThread())){
```

Comment:
better to utilize concurrentHashMap's thread safety with one operation instead of 2:
ThreadInfo i = threadsInfo.get(Thread.currentThread());
if (i != null) { return i.responseHeaders }